packages/tools/devtools/devtools-core/src/data-visualization/DefaultVisualizers.ts

```diff
@@ -103,6 +108,37 @@ export const visualizeSharedCell: VisualizeSharedObject = async (
 	}
 };
 
+/**
+ * Wrapper class for {@link DataObject} to provide a {@link ISharedDirectory} root.
+ * @remarks Intended for devtools internal use only.
+ */
+export class VisualDataObject extends DataObject {
+	public override get root(): ISharedDirectory {
+		return super.root;
+	}
+}
```

Comment:
I still think I would prefer to not have this type, and just cast to some inline type definition. But if we want to keep this, it shouldn't be a class, it should be type-only. We don't actually instantiate one of these things, we just use the type information for casting purposes. E.g.,
```suggestion
/**
 * Wrapper class for {@link DataObject} to provide a {@link ISharedDirectory} root.
 * @remarks Intended for devtools internal use only.
 */
export interface VisualDataObject {
	readonly root: ISharedDirectory;
}
```

Reply:
We should also be very clear in the naming and documentation that this is a hack to get access to `DataObject`'s protected `root` property.